hasjob/views/listing.py

```diff
@@ -560,37 +560,65 @@ def pinnedjob(domain, hashid):
 @app.route('/reject/<hashid>', defaults={'domain': None}, methods=('GET', 'POST'))
 @lastuser.requires_permission('siteadmin')
 def rejectjob(domain, hashid):
+    def send_reject_mail(request, reject_type, post):
+        if reject_type == 'reject':
+            subject = "About your job post on Hasjob"
+            template = 'reject_email.html'
+        elif reject_type == 'ban':
+            subject = "About your account and job posts on Hasjob"
+            template = 'reject_domain_email.html'
+        else:
+            return
+        
+        msg = Message(subject=subject,
+                recipients=[post.email])
+        msg.html = email_transform(render_template(template, post=post), base_url=request.url_root)
+        msg.body = html2text(msg.html)
+        mail.send(msg)
```

Comment:
Oh, I didn't notice this was inside the function. In which case the above bits could be reduced to a dictionary.